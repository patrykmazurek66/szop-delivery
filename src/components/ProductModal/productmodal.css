.product-modal-container {
  width: 800px;
  height: 400px;
  background-color: var(--main50);
  position: absolute;
  top: calc(50vh - 250px);
  left: calc(50vw - 400px);
  color: var(--main900);
  display: flex;
  flex-direction: row;
}

.product-modal-container--product-image {
  width: 350px;
  height: 400px;

  object-fit: cover;
}
.product-modal-container--right-col {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 350px;
  height: 400px;
  margin: 0 50px;
}
.product-modal-container--item-name {
  font-size: 40px;
  font-weight: 300;
  text-transform: capitalize;
  margin: 0;
  text-align: center;
}
.product-modal-container--rating {
  display: flex;
  flex-direction: row;
  align-items: center;
  font-size: 25px;
  margin-top: 20px;
}
.product-modal-container--rating p {
  margin: 0;
}
.product-modal-container--price {
  margin-top: -10px;
  font-size: 20px;
  font-weight: 600;
}
.product-modal-container--description {
  font-size: 15px;
}
.product-modal-container--nutrition-container {
  /* background-color: #ffffff50; */
  font-size: 12px;
  font-weight: 600;
  margin-bottom: 10px;
}
.product-modal-container--nutrition-container h3 {
  text-align: center;
  text-transform: uppercase;
  font-weight: 300;
  margin: 0;
}

.product-modal-container--nutrition-container--item {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 0 5px;
  border-bottom: 1px var(--main300) solid;
}
