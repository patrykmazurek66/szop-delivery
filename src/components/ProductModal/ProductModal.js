import "./productmodal.css";
import StarIcon from "@mui/icons-material/Star";
import StarHalfIcon from "@mui/icons-material/StarHalf";
import StarBorderIcon from "@mui/icons-material/StarBorder";

import { useState, useEffect } from "react";

export default function ProductModal({ name, imgPath, description, rating, price, nutrition }) {
  const [ratingStars, setRatingStars] = useState([]);

  useEffect(() => {
    let tempRating = [];
    for (let i = 0; i < Math.floor(rating); i++) {
      tempRating.push(<StarIcon fontSize="inherit" />);
    }

    Math.floor(rating) < rating ? tempRating.push(<StarHalfIcon fontSize="inherit" />) : {};

    for (let i = Math.ceil(rating); i < 5; i++) {
      tempRating.push(<StarBorderIcon fontSize="inherit" sx={{ color: "var(--main600)" }} />);
    }
    console.log(tempRating, rating);
    setRatingStars(tempRating);
  }, []);

  return (
    <div className="product-modal-container">
      <img src={imgPath} alt={name} className="product-modal-container--product-image" />
      <div className="product-modal-container--right-col">
        <h2 className="product-modal-container--item-name">{name}</h2>
        <div className="product-modal-container--rating">
          {ratingStars}
          <p>({rating})</p>
        </div>
        <p className="product-modal-container--price">${price.toFixed(2)}</p>
        <p className="product-modal-container--description">{description}</p>
        <div className="product-modal-container--nutrition-container">
          <h3>nutrition</h3>
          <div className="product-modal-container--nutrition-container--item">
            <p>Calories:</p>
            <p>{nutrition.calories}</p>
          </div>
          <div className="product-modal-container--nutrition-container--item">
            <p>Fat: </p>
            <p>{nutrition.fat}</p>
          </div>
          <div className="product-modal-container--nutrition-container--item">
            <p>Proteins: </p>
            <p>{nutrition.proteins}</p>
          </div>
          <div className="product-modal-container--nutrition-container--item">
            <p>Carbohydrates: </p>
            <p>{nutrition.carbohydrates}</p>
          </div>
        </div>
      </div>
    </div>
  );
}
