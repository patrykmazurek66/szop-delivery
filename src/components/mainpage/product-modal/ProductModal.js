import "./productmodal.css";

import { useState, useEffect } from "react";
import SetRating from "./SetRating";

export default function ProductModal({ name, imgPath, description, rating, price, nutrition }) {
  const [ratingStars, setRatingStars] = useState([]);

  useEffect(() => {
    setRatingStars(SetRating(rating));
  }, []);

  return (
    <div className="product-modal-container">
      <img
        src={"/food/" + imgPath}
        alt={name}
        className="product-modal-container--product-image"
        loading="lazy"
      />
      <div className="product-modal-container--right-col">
        <h2 className="product-modal-container--item-name">{name}</h2>
        <div className="product-modal-container--rating">
          {ratingStars}
          <p>({rating})</p>
        </div>
        <p className="product-modal-container--price">${(price * 1).toFixed(2)}</p>
        <p className="product-modal-container--description">{description}</p>
        <div className="product-modal-container--nutrition-container">
          <h3>nutrition</h3>
          <div className="product-modal-container--nutrition-container--item">
            <p>Calories:</p>
            <p>{nutrition.calories}</p>
          </div>
          <div className="product-modal-container--nutrition-container--item">
            <p>Fat: </p>
            <p>{nutrition.fat}</p>
          </div>
          <div className="product-modal-container--nutrition-container--item">
            <p>Proteins: </p>
            <p>{nutrition.proteins}</p>
          </div>
          <div className="product-modal-container--nutrition-container--item">
            <p>Carbohydrates: </p>
            <p>{nutrition.carbohydrates}</p>
          </div>
        </div>
      </div>
    </div>
  );
}
