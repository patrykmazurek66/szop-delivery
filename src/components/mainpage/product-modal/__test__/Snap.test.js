import renderer from "react-test-renderer";
import ProductModal from "../ProductModal";

it("Product Modal renders correctly", () => {
  const tree = renderer
    .create(
      <ProductModal
        name={"item.name"}
        description={"item.description"}
        price={"item.price"}
        key={"item.id"}
        imagePath={"blank.png"}
        rating={5}
        nutrition={{
          calories: 520,
          fat: 5,
          proteins: 20,
          carbohydrates: 70,
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
