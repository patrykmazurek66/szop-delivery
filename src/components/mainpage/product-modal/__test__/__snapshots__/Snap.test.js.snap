// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Modal renders correctly 1`] = `
<div
  className="product-modal-container"
>
  <img
    alt="item.name"
    className="product-modal-container--product-image"
    loading="lazy"
    src="/food/undefined"
  />
  <div
    className="product-modal-container--right-col"
  >
    <h2
      className="product-modal-container--item-name"
    >
      item.name
    </h2>
    <div
      className="product-modal-container--rating"
    >
      <p>
        (
        5
        )
      </p>
    </div>
    <p
      className="product-modal-container--price"
    >
      $
      NaN
    </p>
    <p
      className="product-modal-container--description"
    >
      item.description
    </p>
    <div
      className="product-modal-container--nutrition-container"
    >
      <h3>
        nutrition
      </h3>
      <div
        className="product-modal-container--nutrition-container--item"
      >
        <p>
          Calories:
        </p>
        <p>
          520
        </p>
      </div>
      <div
        className="product-modal-container--nutrition-container--item"
      >
        <p>
          Fat: 
        </p>
        <p>
          5
        </p>
      </div>
      <div
        className="product-modal-container--nutrition-container--item"
      >
        <p>
          Proteins: 
        </p>
        <p>
          20
        </p>
      </div>
      <div
        className="product-modal-container--nutrition-container--item"
      >
        <p>
          Carbohydrates: 
        </p>
        <p>
          70
        </p>
      </div>
    </div>
  </div>
</div>
`;
